### PR TITLE
(873) Add an endpoint to get user details

### PR DESCRIPTION
**Pull request by pezholio**
This adds an endpoint to get the user details for a user with a specific ID. If ther details in Delius are different, we update the database to match these details and return the updated object.

### CODE REVIEW

**Comment by pezholio** [comment]
[Confirm if there's enough information in assessment](https://trello.com/c/Jd5xAwWR/873-confirm-if-theres-enough-information-in-assessment)

File: src/main/resources/db/migration/all/20230110115806__add_contact_details_to_user.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE users ADD COLUMN email text;
```
[the remark this answers is not in the document]
Comment (pezholio) [reply src/main/resources/db/migration/all/20230110115806__add_contact_details_to_user.sql:1]: Ooh, yeah, good point

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/UsersController.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.UserTransformer
+
+@Service
+class UsersController(
+  private val userService: UserService,
+  private val userTransformer: UserTransformer
+) : UsersApiDelegate {
+
+  override fun usersIdGet(id: java.util.UUID): ResponseEntity<User> {
```
Comment (brownbearsoftware) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/UsersController.kt:20]: Would move the fully qualified type name here to an import instead - if you hit Alt + Enter (Cmd + Enter on Mac?) then "Add import for 'java.util.UUID'" IntelliJ will do it for you

**Review by brownbearsoftware — APPROVED** [review]
Other than the migration comment, LGTM!

**Pull request merged**